### ISSUE TITLE
reading max double, throws bad conversion

### ISSUE THREAD

**Issue by GoogleCodeExporter** [jbeder/yaml-cpp#229]
```
What steps will reproduce the problem?

#include <fstream>
#include <iostream>
#include <yaml-cpp/yaml.h>
using namespace std;

int main()
{
  double doubleVal;
  float floatVal;
  ofstream outYaml;
  YAML::Node writer;
  YAML::Node reader;

  // write
  writer["float"] = numeric_limits<float>::max();
  writer["double"] = numeric_limits<double>::max();
  outYaml.open("test.yaml");
  outYaml << writer;
  outYaml.close();

  // read
  reader = YAML::LoadFile("test.yaml");
  if (reader["float"])
  {
    floatVal = reader["float"].as<float>();
    cout << "float " << floatVal << endl;
  }
  if (reader["double"])
  {
    doubleVal = reader["double"].as<double>(); // throws bad conversion
    cout << "double " << doubleVal << endl;
  }
  return 0;
}


What is the expected output? What do you see instead?
reading max float is ok, but reading max double throws bad conversion

What version of the product are you using? On what operating system?
Using yaml-cpp 0.5.1 on Ubuntu 12.04 64-bit

Please provide any additional information below.


```

Original issue reported on code.google.com by `lucky13...@gmail.com` on 13 Dec 2013 at 7:36

**Comment by outofthecave** [comment]
I can reproduce this issue on my machine (Ubuntu 16.04 with g++ 5.4.1).  It is indeed not yaml-cpp that fails the parse, but `std::strtod`, which sets `errno` to `ERANGE` for `1.797693134862316e+308`.  This behaviour is correct because `1.797693134862316e+308` is greater than the maximum allowed value for a double, which is `1.7976931348623157e+308`.  (That also explains why it gets converted to infinity on other machines.)

So, the actual problem is that yaml-cpp serializes `1.7976931348623157e+308` as `1.797693134862316e+308`, thereby losing one digit of precision.  This has already been reported as https://github.com/jbeder/yaml-cpp/issues/469.

**Comment by jbeder** [comment]
Fixed by #649.
